HIP-1.md

```diff
@@ -51,18 +51,18 @@ interface IInterchainMessageRecipient {
 }
 
 interface IInterchainSecurityModule {
-    // Used by relayers to determine which off-chain data to inject, and
+    // Used by relayers to determine which off-chain metadata to inject, and
```

Comment:
IDK if i'm using the CCIP-Read hammer the same way I used to do the CIP8 hammer, but should we be using that standard for specifying the retrieval of offchain data in a smart contract?

We don't even need to require RPC services to be run for these ISMs if we are concerned. Instead, we could "shoe-horn"  this by effectively creating a specific URL format which effectively denotes the "module type" as currently specified.

I do think the benefit of CCIP read allows us to decouple the relayer from the offchain data retrieval which seems prudent

Reply:
++

Reply:
I need to re-read up on CCIP-read

Reply:
I think I mostly understand, and agree, in a single ISM context.

@nambrot how would this work for nested ISMs? e.g. an ISM that "ands" together two multisig ISMs

Reply:
I didn't do a deep read but the "Recursive calls in CCIP-aware contracts" section in the EIP may be of use